exonum/src/messages2/mod.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//!
+//! Message represents object received from p2p network.
+//! There next flow between objects:
+//! ```text
+//! +---------+           +---------------+                 +------------+
+//! | Vec<u8> |    ->     | SignedMessage |        ->       |  Protocol  |
+//! +---------+  (verify) +---------------+  (deserialize)  +------------+
+//!                 |                             |
+//!                 V                             V
+//!              (      message dropped if failed       )
+//!
+//! ```
+
+#![allow(missing_docs, missing_debug_implementations)]
+use std::borrow::Cow;
+use std::cmp::PartialEq;
+use std::fmt;
+use std::ops::Deref;
+
+use byteorder::{ByteOrder, LittleEndian};
+use failure::Error;
+
+use crypto::{hash, CryptoHash, Hash, PublicKey};
+use encoding;
+use storage::StorageValue;
+
+use hex::{FromHex, ToHex};
+
+pub(crate) use self::authorization::SignedMessage;
+pub use self::helpers::BinaryForm;
+pub(crate) use self::helpers::{BinaryFormSerialize, HexStringRepresentation};
+pub use self::protocol::*;
+
+#[macro_use]
+mod compatibility;
+mod authorization;
+mod helpers;
+mod protocol;
+#[cfg(test)]
+mod tests;
+
+/// Version of the protocol. Different versions are incompatible.
+pub const PROTOCOL_MAJOR_VERSION: u8 = 1;
+
+/// Transaction raw buffer.
+/// This struct used to transfer transaction in network.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct RawTransaction {
+    service_id: u16,
+    transaction_set: TransactionFromSet,
+}
+
+/// Concrete raw transaction payload inside `TransactionSet` linked with `message_id` in this set.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct TransactionFromSet {
```

Comment:
`TransactionFromSet` is a strange structure name as for me. Can you please explain what is motivate you to chose such name.